### ISSUE TITLE
UnusedFormalParameter should not be used for parameters in inherited methods

### ISSUE THREAD

**Issue by manuelpichler** [phpmd/phpmd#24]
Short description
-----------------

The UnusedFormalParameter rule also marks parameters on methods which are inherited, either from interfaces or parent classes. 

Environment
-----------

n/a

Steps to reproduce
------------------

````php 
interface Foo {
    public function bar($baz, $other);
}

class myFoo implements Foo {
    public function bar($baz, $other)
    {
        // ... some code where $other is not used ...
    }
}
```

Expected behavior
-----------------

As an implementation or a child class is allowed to not use a given parameters the rule should not mark those as violations.

Actual behavior
---------------

Unused parameters on methods declared by an implemented interface or overwritten methods of parent classes are recorded as violations.

[Synced from PivotalTracker: 'http://www.pivotaltracker.com/story/show/9355837']

**Comment by ava-dylang** [comment]
:+1:

**Comment by ravage84** [comment]
Will be worked on in https://github.com/phpmd/phpmd/issues/180
Closing
